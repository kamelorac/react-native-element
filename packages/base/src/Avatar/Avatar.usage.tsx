import React from 'react';
import { Avatar } from '..';
import { usage, Stack, info, meta } from '@rneui/doc-gen';

info(
  'Avatars are found all over ui design from lists to profile screens.',
  'They are commonly used to represent a user and can contain photos, icons, or even text.'
);

meta({
  name: 'Avatar',
});

usage('Shape', '', () => (
  <Stack row spacing={4}>
    <Avatar
      size={32}
      rounded
      icon={{ name: 'pencil', type: 'font-awesome' }}
      containerStyle={{ backgroundColor: '#9700b9' }}
    />
    <Avatar
      size={32}
      icon={{ name: 'pencil', type: 'font-awesome' }}
      containerStyle={{ backgroundColor: '#9700b9' }}
    />
  </Stack>
));

usage('Sizes', '', () => (
  <Stack row spacing={4}>
    <Avatar
      size={24}
      rounded
      icon={{ name: 'pencil', type: 'font-awesome' }}
      containerStyle={{ backgroundColor: '#9700b9' }}
    />
    <Avatar
      size={32}
      rounded
      icon={{ name: 'pencil', type: 'font-awesome' }}
      containerStyle={{ backgroundColor: '#9700b9' }}
    />
    <Avatar
      size={48}
      rounded
      icon={{ name: 'pencil', type: 'font-awesome' }}
      containerStyle={{ backgroundColor: '#9700b9' }}
    />
  </Stack>
));
usage('Image Avatar', '', () => (
  <Stack row spacing={4}>
    <Avatar
      size={32}
      rounded
      source={{ uri: 'https://randomuser.me/api/portraits/men/36.jpg' }}
    />
    <Avatar
      size={32}
      rounded
      source={{ uri: 'https://randomuser.me/api/portraits/men/35.jpg' }}
    />
  </Stack>
));

usage('Letter Avatar', '', () => (
  <Stack row spacing={4}>
    <Avatar
      size={32}
      rounded
      title="Rd"
      containerStyle={{ backgroundColor: 'blue' }}
    />
    <Avatar
      size={32}
      rounded
      title="AB"
      containerStyle={{ backgroundColor: 'purple' }}
    />
  </Stack>
));
