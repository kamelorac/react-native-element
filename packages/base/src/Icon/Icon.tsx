import React from 'react';
import {
  Platform,
  View,
  StyleSheet,
  ViewStyle,
  StyleProp,
  TextStyle,
  Pressable,
  ColorValue,
} from 'react-native';
import {
  IconButtonProps,
  IconProps as VectorIconProps,
} from 'react-native-vector-icons/Icon';
import Color from 'color';
import getIconType from '../helpers/getIconType';
import getIconStyle from '../helpers/getIconStyle';
import {
  androidRipple,
  defaultTheme,
  InlinePressableProps,
  RneFunctionComponent,
} from '../helpers';

export type IconType =
  | 'material'
  | 'material-community'
  | 'simple-line-icon'
  | 'zocial'
  | 'font-awesome'
  | 'octicon'
  | 'ionicon'
  | 'foundation'
  | 'evilicon'
  | 'entypo'
  | 'antdesign'
  | 'font-awesome-5'
  | string;

export interface IconObject {
  /** Name of icon. */
  name?: string;

  /** Color of icon. */
  color?: string;

  /** Size of icon. */
  size?: number;

  /** Type of icon */
  type?: IconType;

  /** Apply style to the icon using iconStyle. */
  iconStyle?: StyleProp<TextStyle>;
}

export type IconNode = boolean | React.ReactElement<{}> | Partial<IconProps>;

export interface IconProps extends InlinePressableProps, IconButtonProps {
  /** Type of icon set. [Supported sets here](#available-icon-sets). */
  type?: IconType;

  /** Update React Native Component.
   *  @default `Press handlers present then Pressable else View`
   */
  Component?: typeof React.Component;

  /** Reverses color scheme. */
  reverse?: boolean;

  /** Adds box shadow to button. */
  raised?: boolean;

  /** Add styling to container holding icon. */
  containerStyle?: StyleProp<ViewStyle>;

  /** Provide all props from react-native Icon component. */
  iconProps?: VectorIconProps;

  /** Specify reverse icon color. */
  reverseColor?: string;

  /** Disables onPress events. Only works when `onPress` has a handler. */
  disabled?: boolean;

  /** Style for the button when disabled. Only works when `onPress` has a handler. */
  disabledStyle?: StyleProp<ViewStyle>;

  /** Uses the solid font. */
  solid?: boolean;

  /** Uses the brands font (FontAwesome5 only). */
  brand?: boolean;
}

/** Icons are visual indicators usually used to describe action or intent.
 * They are also used for displaying information. */
export const Icon: RneFunctionComponent<IconProps> = ({
  type = 'material',
  name,
  size = 24,
  color: colorProp,
  iconStyle,
  iconProps,
  underlayColor = 'transparent',
  reverse = false,
  raised = false,
  containerStyle,
  reverseColor: reverseColorProp,
  disabled = false,
  disabledStyle,
  onPress,
  onLongPress,
  onPressIn,
  onPressOut,
  Component = onPress || onLongPress || onPressIn || onPressOut
    ? Pressable
    : View,
  solid = false,
  brand = false,
  theme = defaultTheme,
  pressableProps,
  ...rest
}) => {
  const color = colorProp || theme?.colors?.black;
  const reverseColor = reverseColorProp || theme?.colors?.white;
  const IconComponent = getIconType(type);
  const iconSpecificStyle = getIconStyle(type, { solid, brand });

  const getBackgroundColor = React.useMemo(() => {
    if (reverse) {
      return color;
    }
    return raised ? theme?.colors?.white : 'transparent';
  }, [color, raised, reverse, theme?.colors?.white]);

  const buttonStyles = React.useMemo(
    () => ({
      borderRadius: size + 4,
      height: size * 2 + 4,
      width: size * 2 + 4,
    }),
    [size]
  );

  return (
    <View
      style={StyleSheet.flatten([
        !raised && styles.container,
        (reverse || raised) && styles.button,
        (reverse || raised) && buttonStyles,
        raised && styles.raised,
        iconStyle && iconStyle.borderRadius
          ? {
              borderRadius: iconStyle.borderRadius,
            }
          : {},
        containerStyle && containerStyle,
      ])}
      testID="RNE__ICON__CONTAINER"
    >
      <Component
        testID="RNE__ICON__CONTAINER_ACTION"
        {...{
          android_ripple: androidRipple(
            Color(reverse ? color : (underlayColor as string))
              .alpha(0.3)
              .rgb()
              .string()
          ),
          onPress,
          onLongPress,
          onPressIn,
          onPressOut,
          disabled,
          accessibilityRole: 'button',
          ...pressableProps,
          ...rest,
        }}
      >
        <View
          style={StyleSheet.flatten([
            (reverse || raised) && buttonStyles,
            {
              backgroundColor: getBackgroundColor as ColorValue,
              alignItems: 'center',
              justifyContent: 'center',
            },
            disabled && styles.disabled,
            disabled && disabledStyle,
          ])}
          testID="RNE__ICON"
        >
          <IconComponent
            testID="RNE__ICON__Component"
            style={StyleSheet.flatten([
              { backgroundColor: 'transparent' },
              iconStyle && iconStyle,
            ])}
            size={size}
            name={name}
            color={reverse ? reverseColor : color}
            {...iconSpecificStyle}
            {...iconProps}
          />
        </View>
      </Component>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    overflow: 'hidden',
  },
  button: {
    margin: 7,
  },
  raised: {
    ...Platform.select({
      android: {
        elevation: 2,
      },
      default: {
        shadowColor: 'rgba(0,0,0, .4)',
        shadowOffset: { height: 1, width: 1 },
        shadowOpacity: 1,
        shadowRadius: 1,
      },
    }),
  },
  disabled: {
    backgroundColor: '#D1D5D8',
  },
});

Icon.displayName = 'Icon';
