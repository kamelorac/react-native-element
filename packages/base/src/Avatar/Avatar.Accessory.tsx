import React from 'react';
import {
  Pressable,
  View,
  Platform,
  StyleSheet,
  StyleProp,
  ViewStyle,
  ColorValue,
} from 'react-native';
import { ImageProps, Image } from '../Image';
import { IconProps, Icon } from '../Icon';
import {
  androidRipple,
  InlinePressableProps,
  RneFunctionComponent,
} from '../helpers';

export type AccessoryProps = Partial<IconProps> &
  Partial<ImageProps> &
  InlinePressableProps & {
    /** Add underlay color to the accessory of avatar. */
    underlayColor?: ColorValue;

    /** Add custom styling to the accessory of avatar. */
    style?: StyleProp<ViewStyle>;
  };

/** This is used for adding an accessory to the Avatar.
 * Receives either all [Icon](icon#props) or [Image](image#props) props. */
export const Accessory: RneFunctionComponent<AccessoryProps> = ({
  size = 10,
  style,
  underlayColor = '#000',
  onPress,
  onLongPress,
  onPressIn,
  onPressOut,
  source,
  pressableProps,
  ...rest
}: AccessoryProps) => {
  return (
    <Pressable
      {...{
        android_ripple:
          (onPress || onLongPress) && androidRipple(underlayColor),
        ...pressableProps,
      }}
      style={[
        styles.accessory,
        {
          width: size,
          height: size,
          borderRadius: size / 2,
        },
        style,
      ]}
      {...{ onPressOut, onPressIn, onPress, onLongPress }}
    >
      <View>
        {source ? (
          <Image
            source={source}
            style={{
              width: size,
              height: size,
              borderRadius: size / 2,
            }}
            {...rest}
          />
        ) : (
          <Icon
            name="mode-edit"
            type="material"
            color="#fff"
            size={size * 0.8}
            {...rest}
          />
        )}
      </View>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  accessory: {
    position: 'absolute',
    bottom: 0,
    right: 0,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#aaa',
    ...Platform.select({
      android: {
        elevation: 1,
      },
      default: {
        shadowColor: '#000',
        shadowOffset: { width: 1, height: 1 },
        shadowRadius: 2,
        shadowOpacity: 0.5,
      },
    }),
  },
});

Accessory.displayName = 'Avatar.Accessory';
