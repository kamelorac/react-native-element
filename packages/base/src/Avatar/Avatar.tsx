import React from 'react';
import {
  View,
  Text,
  Image as RNImage,
  StyleSheet,
  Pressable,
  StyleProp,
  ViewStyle,
  TextStyle,
  ImageSourcePropType,
  ImageURISource,
  ImageStyle,
} from 'react-native';
import {
  InlinePressableProps,
  renderNode,
  RneFunctionComponent,
} from '../helpers';
import { IconObject, Icon } from '../Icon';
import { ImageProps, Image } from '../Image';

export const avatarSizes = {
  small: 34,
  medium: 50,
  large: 75,
  xlarge: 150,
};

type AvatarIcon = IconObject & {
  iconStyle?: StyleProp<TextStyle>;
};

export interface AvatarProps extends InlinePressableProps {
  /** Component for enclosing element (eg: TouchableHighlight, View, etc).
   *
   *  @default `Press handlers present then Pressable else View`
   */
  Component?: typeof React.Component;

  /** Callback function when pressing component. */
  onPress?(): void;

  /** Callback function when long pressing component. */
  onLongPress?(): void;

  /** Styling for outer container. */
  containerStyle?: StyleProp<ViewStyle>;

  /** Image source to be displayed on avatar. */
  source?: ImageSourcePropType;

  /** Style for avatar image. */
  avatarStyle?: ImageStyle;

  /** Makes the avatar circular. */
  rounded?: boolean;

  /** Renders title in the placeholder. */
  title?: string;

  /** Style for the title. */
  titleStyle?: StyleProp<TextStyle>;

  /** Style for the view outside image or icon. */
  overlayContainerStyle?: StyleProp<TextStyle>;

  /** Displays an icon as the main content of the Avatar. **Cannot be used alongside title**. When used with the `source` prop it will be used as the placeholder. */
  icon?: AvatarIcon;

  /** Extra styling for icon component. */
  iconStyle?: StyleProp<TextStyle>;

  /** Size of the avatar. */
  size?: ('small' | 'medium' | 'large' | 'xlarge') | number;

  /** Adds style to the placeholder wrapper. */
  placeholderStyle?: StyleProp<ViewStyle>;

  /** Custom placeholder element (by default, it's the title). */
  renderPlaceholderContent?: React.ReactElement<{}>;

  /** Optional properties to pass to the avatar e.g "resizeMode". */
  imageProps?: Partial<ImageProps>;

  /** Custom ImageComponent for Avatar. */
  ImageComponent?: React.ComponentClass;
}

/**
 * Avatars are found all over ui design from lists to profile screens.
 * They are commonly used to represent a user and can contain photos, icons, or even text.

```
 * */
export const Avatar: RneFunctionComponent<AvatarProps> = ({
  onPress,
  onLongPress,
  onPressIn,
  onPressOut,
  Component = onPress || onLongPress || onPressIn || onPressOut
    ? Pressable
    : View,
  containerStyle,
  icon,
  iconStyle,
  source,
  size = 'small',
  avatarStyle,
  rounded,
  title,
  titleStyle,
  overlayContainerStyle,
  imageProps,
  placeholderStyle,
  renderPlaceholderContent,
  ImageComponent = RNImage,
  children,
  pressableProps,
  ...rest
}) => {
  const width =
    typeof size === 'number' ? size : avatarSizes[size] || avatarSizes.small;

  const height = width;
  const titleSize = width / 2;
  const iconSize = width / 2;

  const PlaceholderContent =
    (renderPlaceholderContent &&
      renderNode(undefined, renderPlaceholderContent)) ||
    (title && (
      <Text
        style={StyleSheet.flatten([
          styles.title,
          { fontSize: titleSize },
          titleStyle,
        ])}
      >
        {title}
      </Text>
    )) ||
    (icon && (
      <Icon
        style={StyleSheet.flatten([iconStyle && iconStyle])}
        color={icon.color || 'white'}
        name={icon.name || 'account'}
        size={icon.size || iconSize}
        type={icon.type || 'material-community'}
      />
    ));

  const hidePlaceholder = !(source && (source as ImageURISource).uri);

  const imageContainerStyle = StyleSheet.flatten([
    styles.overlayContainer,
    rounded && { borderRadius: width / 2, overflow: 'hidden' },
    overlayContainerStyle,
    imageProps && imageProps.containerStyle,
  ]);
  if (imageProps && imageProps.containerStyle) {
    delete imageProps.containerStyle;
  }

  return (
    <Component
      style={StyleSheet.flatten([
        styles.container,
        { height, width },
        rounded && { borderRadius: width / 2 },
        containerStyle,
      ])}
      {...{
        onPress,
        onLongPress,
        onPressIn,
        onPressOut,
        ...pressableProps,
        ...rest,
      }}
    >
      <Image
        testID="RNE__Avatar__Image"
        placeholderStyle={StyleSheet.flatten([
          placeholderStyle,
          hidePlaceholder && styles.hiddenPlaceholderStyle,
        ])}
        PlaceholderContent={PlaceholderContent}
        containerStyle={imageContainerStyle as StyleProp<TextStyle>}
        source={source}
        borderRadius={rounded ? width / 2 : undefined}
        {...imageProps}
        style={StyleSheet.flatten([
          styles.avatar,
          imageProps && imageProps.style,
          avatarStyle,
        ])}
        ImageComponent={ImageComponent}
      />
      {children}
    </Component>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'transparent',
  },
  avatar: {
    flex: 1,
    width: undefined,
    height: undefined,
  },
  overlayContainer: {
    flex: 1,
  },
  title: {
    color: '#ffffff',
    backgroundColor: 'transparent',
    textAlign: 'center',
  },
  hiddenPlaceholderStyle: {
    backgroundColor: 'transparent',
  },
});

Avatar.displayName = 'Avatar';
