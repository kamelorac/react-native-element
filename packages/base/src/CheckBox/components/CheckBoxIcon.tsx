import React from 'react';
import { RneFunctionComponent } from '../../helpers';
import getIconType from '../../helpers/getIconType';
import { IconType } from '../../Icon';

export interface CheckBoxIconProps {
  /** Flag for checking the icon. */
  checked: boolean;

  /** onPress function for checkbox. */
  onIconPress?(): void;

  /** onLongPress function for checkbox. */
  onLongIconPress?(): void;

  /** Size of the checkbox. */
  size?: number;

  /** Default checked icon ([Font Awesome Icon](http://fontawesome.io/icons/)). */
  checkedIcon?: string | React.ReactElement<{}>;

  /** Default checked icon ([Font Awesome Icon](http://fontawesome.io/icons/)) */
  uncheckedIcon?: string | React.ReactElement<{}>;

  /** Type of icon set. [Supported sets here](icon#available-icon-sets). */
  iconType?: IconType;

  /** Default checked color. */
  checkedColor?: string;

  /** Default unchecked color. */
  uncheckedColor?: string;
}

export const CheckBoxIcon: RneFunctionComponent<CheckBoxIconProps> = ({
  checked,
  onIconPress,
  onLongIconPress,
  size = 24,
  checkedIcon = 'check-square-o',
  uncheckedIcon = 'square-o',
  iconType,
  checkedColor,
  uncheckedColor = '#bfbfbf',
}) => {
  if (checked && React.isValidElement(checkedIcon)) {
    return checkedIcon;
  }
  if (!checked && React.isValidElement(uncheckedIcon)) {
    return uncheckedIcon;
  }
  const VectorIcon = iconType
    ? getIconType(iconType)
    : getIconType('font-awesome');
  return (
    <VectorIcon
      testID="RNE__Checkbox__Icon"
      color={checked ? checkedColor : uncheckedColor}
      name={checked ? checkedIcon : uncheckedIcon}
      size={size || 24}
      style={{ minWidth: size || 24 }}
      onLongPress={onLongIconPress}
      onPress={onIconPress}
    />
  );
};
