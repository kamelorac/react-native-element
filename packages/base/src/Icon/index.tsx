import { Icon, IconProps, IconNode, IconObject, IconType } from './Icon';

export { Icon };
export type { IconProps, IconNode, IconObject, IconType };
