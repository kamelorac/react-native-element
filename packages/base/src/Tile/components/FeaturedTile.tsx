import React from 'react';
import {
  Pressable,
  View,
  StyleSheet,
  Dimensions,
  StyleProp,
  TextStyle,
} from 'react-native';
import { BackgroundImage } from '../../helpers';
import { renderNode, RneFunctionComponent } from '../../helpers';
import { Text } from '../../Text';
import { Icon } from '../../Icon';
import { TileProps } from '../Tile';

const renderText = (
  content: React.ReactNode,
  defaultProps: any,
  style: StyleProp<TextStyle>
) =>
  renderNode(Text, content, {
    ...defaultProps,
    style: StyleSheet.flatten([style, defaultProps && defaultProps.style]),
  });

const FeaturedTile: RneFunctionComponent<TileProps> = ({
  title,
  icon,
  caption,
  imageSrc,
  containerStyle,
  imageContainerStyle,
  overlayContainerStyle,
  iconContainerStyle,
  titleStyle,
  captionStyle,
  ImageComponent = BackgroundImage,
  imageProps = {},
  width = Dimensions.get('window').width,
  height = width * 0.8,
  ...attributes
}) => {
  return (
    <Pressable
      {...attributes}
      style={StyleSheet.flatten([
        { width, height },
        containerStyle && containerStyle,
      ])}
    >
      <ImageComponent
        resizeMode="cover"
        {...imageProps}
        source={imageSrc}
        style={StyleSheet.flatten([
          { width, height },
          styles.imageContainer,
          imageContainerStyle && imageContainerStyle,
        ])}
      >
        <View
          style={StyleSheet.flatten([
            styles.overlayContainer,
            overlayContainerStyle && overlayContainerStyle,
          ])}
        >
          <View
            style={StyleSheet.flatten([
              styles.iconContainer,
              iconContainerStyle && iconContainerStyle,
            ])}
          >
            {icon && <Icon {...icon} />}
          </View>
          <Text
            testID="featuredTileTitle"
            h4={!titleStyle || !('fontSize' in titleStyle)}
            style={StyleSheet.flatten([styles.text, titleStyle && titleStyle])}
          >
            {title}
          </Text>
          {renderText(caption, { style: captionStyle }, styles.text)}
        </View>
      </ImageComponent>
    </Pressable>
  );
};

export { FeaturedTile };

const styles = StyleSheet.create({
  imageContainer: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  overlayContainer: {
    flex: 1,
    alignItems: 'center',
    alignSelf: 'stretch',
    justifyContent: 'center',
    paddingLeft: 25,
    paddingRight: 25,
    paddingTop: 45,
    paddingBottom: 40,
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  text: {
    color: '#ffffff',
    backgroundColor: 'rgba(0,0,0,0)',
    marginBottom: 15,
    textAlign: 'center',
  },
  iconContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
  },
});
