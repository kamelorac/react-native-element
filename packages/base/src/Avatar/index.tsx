import { Avatar as AvatarBase, AvatarProps } from './Avatar';
import { Accessory, AccessoryProps } from './Avatar.Accessory';

const Avatar = Object.assign(AvatarBase, {
  Accessory,
});
export { Avatar, Accessory };

export type { AccessoryProps, AvatarProps };
