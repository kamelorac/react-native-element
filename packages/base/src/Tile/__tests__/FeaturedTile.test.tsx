import React from 'react';
import { Avatar } from '../../Avatar';
import { FeaturedTile } from '../components/FeaturedTile';
import { renderWithWrapper } from '../../../.ci/testHelper';
import { Pressable, Image } from 'react-native';
import { Icon } from '../../Icon';

describe('FeaturedTitle component', () => {
  it('should match snapshot', () => {
    const component = renderWithWrapper(
      <FeaturedTile imageSrc={{ uri: 'http://google.com' }} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with width and height', () => {
    const { wrapper } = renderWithWrapper(
      <FeaturedTile
        imageSrc={{ uri: 'http://google.com' }}
        width={34}
        height={20}
      />
    );
    expect(wrapper.findByType(Pressable).props.style).toMatchObject({
      width: 34,
      height: 20,
    });
  });

  it('should render with Icon', () => {
    const { wrapper } = renderWithWrapper(
      <FeaturedTile
        imageSrc={{ uri: 'http://googleRNE__ICON__CONTAINER.com' }}
        icon={{ name: 'play-circle', type: 'font-awesome' }}
        imageContainerStyle={{ height: 70 }}
        containerStyle={{ height: 70 }}
        captionStyle={{ color: 'blue' }}
        iconContainerStyle={{ height: 70 }}
        titleStyle={{ backgroundColor: 'yellow' }}
        overlayContainerStyle={{ height: 70 }}
      />
    );
    expect(wrapper.findByType(Icon).props).toMatchObject({
      name: 'play-circle',
      type: 'font-awesome',
    });
  });

  it('should render component in caption', () => {
    const { wrapper } = renderWithWrapper(
      <FeaturedTile
        imageSrc={{ uri: 'http://google.com' }}
        caption={<Avatar source={{ uri: 'http://google.com' }} />}
      />
    );
    expect(wrapper.findByType(Avatar).props).toMatchObject({
      source: { uri: 'http://google.com' },
    });
  });

  it('should apply custom image props', () => {
    const { wrapper } = renderWithWrapper(
      <FeaturedTile
        imageSrc={{ uri: 'http://google.com' }}
        imageProps={{ resizeMode: 'contain' }}
      />
    );
    expect(wrapper.findByType(Image).props).toMatchObject({
      resizeMode: 'contain',
    });
  });

  it('should render string in caption', () => {
    const { queryByText } = renderWithWrapper(
      <FeaturedTile
        imageSrc={{ uri: 'http://google.com' }}
        caption="Caption text"
      />
    );
    expect(queryByText('Caption text')).not.toBeNull();
  });
});
