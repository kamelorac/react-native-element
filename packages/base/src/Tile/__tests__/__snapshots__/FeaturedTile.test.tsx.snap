// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedTitle component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "height": 600,
        "width": 750,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "alignItems": "center",
          "height": 600,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          {
            "uri": "http://google.com",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": 600,
              "width": 750,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "bottom": 0,
            "flex": 1,
            "justifyContent": "center",
            "left": 0,
            "paddingBottom": 40,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 45,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "alignSelf": "center",
              "justifyContent": "center",
            }
          }
        />
        <Text
          accessibilityRole="text"
          style={
            {
              "backgroundColor": "rgba(0,0,0,0)",
              "color": "#ffffff",
              "fontSize": 28.285714285714285,
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
          testID="featuredTileTitle"
        />
      </View>
    </View>
  </View>
</View>
`;
